lib/internal/test_runner/runner.js

```diff
@@ -319,7 +349,21 @@ class FileTest extends Test {
 
 function runTestFile(path, filesWatcher, opts) {
   const watchMode = filesWatcher != null;
-  const subtest = opts.root.createSubtest(FileTest, path, { __proto__: null, signal: opts.signal }, async (t) => {
+  const Factory = opts.isolation === 'none' ? InProcessFileTest : SpawnFileTest;
+  const subtest = opts.root.createSubtest(Factory, path, { __proto__: null, signal: opts.signal }, async (t) => {
+    if (opts.isolation === 'none') {
+      let parentURL;
+      try {
+        parentURL = pathToFileURL(process.cwd() + '/').href;
+      } catch {
+        parentURL = 'file:///';
+      }
```

Comment:
You can use `const { getCWDURL } = require('internal/util')` from utils to avoid recreating the `URL` instance if it was already computed.
```suggestion
      const parentURL = getCWDURL().href;
```